numba/cpython/randomimpl.py

```diff
@@ -1586,3 +1586,61 @@ def dirichlet_impl(alpha, size=None):
         )
 
     return dirichlet_impl
+
+# ------------------------------------------------------------------------
+# np.random.noncentral_chisquare
+
+@overload(np.random.noncentral_chisquare)
+def noncentral_chisquare(df, nonc, size=None):
+
+    @register_jitable
+    def noncentral_chisquare_single(df, nonc):
+        # identical to numpy implementation from distributions.c
+        # https://github.com/numpy/numpy/blob/c65bc212ec1987caefba0ea7efe6a55803318de9/numpy/random/src/distributions/distributions.c#L797
+        
+        if df <= 0:
+            raise ValueError("df <= 0")
+        if nonc < 0:
+            raise ValueError("nonc < 0")
+
+        if np.isnan(nonc):
+            return np.nan
+
+        if 1 < df:
+            chi2 = np.random.chisquare(df-1)
+            n = np.random.standard_normal() + np.sqrt(nonc)
+            return chi2 + n * n
+
+        else:
+            i = np.random.poisson(nonc/2.0)
+            return np.random.chisquare(df + 2 * i)
+
+    if size in (None, types.none):
+        def noncentral_chisquare_impl(df, nonc, size=None):
+            return noncentral_chisquare_single(df, nonc)
+
+    elif isinstance(size, types.Integer):
+
+        def noncentral_chisquare_impl(df, nonc, size=None):
+            out = np.empty(size)
+            out_flat = out.flat
+            for idx, _ in enumerate(out_flat):
```

Comment:
I have two questions about this loop:

a) Since we only need the indices anyway, can we use `for i in range(size)`
b) It look as though `out_flat` will receive identical values? Is this correct? Can we try to use broadcasting for the assignment?

Reply:
Thank you for your comments. 
a) Yes, I've made it so. 
b) Each element will be a different realization of the RNG, so unfortunately, no. I tried to rewrite it so that it could take advantage of the fact that its underlying distributions can take size arguments, but it did not work so well, as none of the current random distribution implementations can handle arrays for their location and scale arguments.

Reply:
OK, thank you for clearning this up, should be fine then.